src/param.c

```diff
@@ -142,6 +142,63 @@ stumpless_new_param( const char *name, const char *value ) {
   return result;
 }
 
+struct stumpless_param *
+stumpless_new_param_from_string( const char *string ) {
+
+  VALIDATE_ARG_NOT_NULL( string );
+
+  int i;
+  int name_len = 0, name_start = 0, value_len = 0, value_start = 0;
+
+  /* Check that the characters in 'name' are allowed. */
+  for (i = 0; string[i] != '='; i++) {
+    if (!((string[i] >= 'a' && string[i] <= 'z') ||
+	  (string[i] >= 'A' && string[i] <= 'Z') ||
+	  (string[i] >= '0' && string[i] <= '9') ||
+	  (string[i] == '-') ||
+	  (string[i] == '.') ||
+	  (string[i] == '_'))){
+      raise_invalid_param(  );
+      return NULL;
+    }
+  }
+  /* Check if the character after the '=' is '"', else raise an error. */
+  if (string[i + 1] != '"'){
+    raise_invalid_param(  );
+    return NULL;
+  }
+
+  name_len = i;
+  /* Declare an array to hold the 'name' string.
+     We add 1 to the length for the '\0' character. */
+  char name[name_len + 1];
+  /* We make this addition to skip '=' and '"' characters. */
+  i += 2;
+  value_start = i;
+
+  /* Iterate to the end of the 'value' string. */
+  for (; string[i] != '\0'; i++){}
+  /* If the final character isn't '"' we raise an error. */
+  if (string[i - 1] != '"') {
+    raise_invalid_param(  );
+    return NULL;
+  } else {
+    /* We make this substraction for the 2 '"' characters and for the '=' character. */
+    value_len = i - name_len - 3;
+  }
+  
+  /* Declare an array to hold the 'value' string.
+     We add 1 to the length for the '\0' character. */
+  char value[value_len + 1];
+  
+  strncpy(name, string + name_start, name_len);
```

Comment:
`memcpy` is a better choice here, as you already know the length. This may also correct the Windows builds that are currently failing.